src/entities/pools/weighted/weightedJoin.ts

```diff
@@ -0,0 +1,187 @@
+import { BaseJoin, JoinInput, JoinQueryResult, PoolState } from '..';
```

Comment:
Looks good so far I think. Probably just need to keep in mind when building it out how we can keep it modular so we can reuse when it comes to chaining as part of a relayer call.

Reply:
I'll try to keep that in mind, but I feel like we'll be able to better identify what needs to be adapted only when we add chaining/relayer calls.